pkgs/by-name/op/openstreetmap-carto/package.nix

```diff
@@ -0,0 +1,210 @@
+{
+  carto,
+  fetchFromGitHub,
+  fetchurl,
+  gdal,
+  hanazono,
+  lib,
+  nix,
+  nixosTests,
+  noto-fonts,
+  python3,
+  runCommand,
+  stdenvNoCC,
+  symlinkJoin,
+  writeShellApplication,
+}:
+
+let
+  generate_env = python3.withPackages (
+    p: with p; [
+      colormath
+      lxml
+      pyyaml
+    ]
+  );
+  get-external-data_env = python3.withPackages (
+    p: with p; [
+      psycopg2
+      pyyaml
+      requests
+    ]
+  );
+
+  mkArchive =
+    {
+      url,
+      archive_time,
+      hash,
+    }:
+    {
+      inherit url;
+      archive = fetchurl {
+        url = "https://web.archive.org/web/${archive_time}/${url}";
+        inherit hash;
+      };
+    };
+  simplified_water_polygons = mkArchive {
```

Comment:
What about moving all data downloading code to `nixos/tests/nik4.nix` test and generate `external-data.yml` file required for test script there ? It looks to me as a more appropriate place.

Reply:
My goal was to make this script as reproducible as possible, no matter if it's called in the NixOS test or on a users system. I'm in favor of keeping it this way.

Reply:
What is the purpose of this data ? Isn't it useful mostly for testing ? If that's true, I still lean more towards moving this code to the test.
Another issue of having this code in package is how to update this data over the time (and how often). We might end up with reproducible but very outdated data.

What do you think @autra ?